content/guides/getstarted/setup-local-development.md

```diff
@@ -38,48 +38,266 @@ deploying Solana programs to the blockchain.
+
+In macOS build tools are given by Xcode command line tools. You can download it from this link. 
+Sign in with your apple Id and download. 
+
+You can check if your xcode cli is installed via this command.
+
+```jsx
+xcode-select -p
+```
+
+If you don’t see a path returned, you need to install the cli tools.
+
+There are three ways to install xcode cli tools
+
+1. Install tools via terminal
+    
+    ```jsx
+    xcode-select --install
+    ```
+    
+2. Download installer and install it with graphical interface
+    
+    [](https://developer.apple.com/download/all/)
+    
+    ![setup—-xcode](/assets/guides/setup-local-environment/setup—-xcode.png)
+    
+3. You might also like to install homebrew to install packages like linux people do using apt-get.
+Optional way to install homebrew and Xcode build utilities is to follow the below blog.
+    
+    [Xcode Command Line Tools with Homebrew · Mac Install Guide](https://mac.install.guide/commandlinetools/3)
+    
+
+---
+
+Congrats, we have installed system dependencies and build essentials required to build software.
+
+---
+
+## 2. Install rust
+
+Rust is a multi-paradigm, general-purpose programming language that emphasizes performance, type safety, and concurrency. Or let’s just say its a coool language in boomer world.
+
+rustc (compiler for rust), cargo (package manager for rust) and rustup (rust version installer & manager for stable and beta versions) all three will setup at once.
+
+### Install Rust for Windows
+
+Go to the below url, click on install ***rustup-init.exe***, this will start installing rust for you.
+
+Make sure to restart your computer after installation just in case to load all the changes in current environment.
+
+![setup—-rustup-windows](/assets/guides/setup-local-environment/setup—-rustup-windows.png)
+
+[Other Installation Methods - Rust Forge](https://forge.rust-lang.org/infra/other-installation-methods.html#other-ways-to-install-rustup)
+
+### Install Rust for macOS, Linux, WSL or another Unix-like OS.
+
+one single spell will set you up.
+It downloads the script to setup in your computer and runs the script to install binaries.
+
+```jsx
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+Now either refresh your terminal, or just close and open a new one.
+
+```jsx
+source ~/.bashrc
 ```
 
-## Setup a localhost blockchain cluster
+---
+
+## 3. Install Solana CLI
+
+To interact with a Solana cluster (bunch of big state synced servers), we will use its command-line interface, also known as the CLI. We use the command-line because it is the first place the Solana core team deploys new functionality. The command-line interface is not necessarily the easiest to use, but it provides the most direct, flexible, and secure access to your Solana accounts (storage units on solana).
+
+### for windows
+
+1. Open terminal as administrator, copy and paste this command in your cmd, this downloads solana toolchain into a temporary location on C drive. 
```

Comment:
Windows uses powershell as the default shell, also I think Windows may include curl now. Not sure `cmd` is necessary.